coderd/rbac/USAGE.md

```diff
@@ -0,0 +1,344 @@
+| **userAdmin**        | Administrator of users                                              | users, groups, role assignments              |
+| **orgAdmin**         | Like **owner**, but scoped to a single organization                 | _(org-level equivalent)_                     |
+| **orgMember**        | Like **member**, but scoped to a single organization                | _(org-level equivalent)_                     |
+| **orgAuditor**       | Like **auditor**, but scoped to a single organization               | _(org-level equivalent)_                     |
+| **orgUserAdmin**     | Like **userAdmin**, but scoped to a single organization             | _(org-level equivalent)_                     |
+| **orgTemplateAdmin** | Like **templateAdmin**, but scoped to a single organization         | _(org-level equivalent)_                     |
+
+_\* except some, which are not important to this overview_
+
+## Actions
+
+Roles are collections of permissions (we call them _actions_).
+
+These can be found in `coderd/rbac/policy/policy.go`.
+
+| Action                  | Description                             |
+| ----------------------- | --------------------------------------- |
+| **create**              | Create a resource                       |
+| **read**                | Read a resource                         |
+| **update**              | Update a resource                       |
+| **delete**              | Delete a resource                       |
+| **use**                 | Use a resource                          |
+| **read_personal**       | Read owned resource                     |
+| **update_personal**     | Update owned resource                   |
+| **ssh**                 | SSH into a workspace                    |
+| **application_connect** | Connect to workspace apps via a browser |
+| **view_insights**       | View deployment insights                |
+| **start**               | Start a workspace                       |
+| **stop**                | Stop a workspace                        |
+| **assign**              | Assign user to role / org               |
+
+# Creating a new noun
+
+In the following example, we're going to create a new RBAC noun for a new entity
+called a "frobulator" _(just some nonsense word for demonstration purposes)_.
+
+_Refer to https://github.com/coder/coder/pull/14055 to see a full
+implementation._
+
+## Creating a new entity
+
+If you're creating a new resource which has to be owned by users of differing
+roles, you need to create a new RBAC resource.
+
+Let's say we're adding a new table called `frobulators` (we'll use this table
+later):
+
+```sql
+CREATE TABLE frobulators
+(
+  id           uuid NOT NULL,
+  user_id      uuid NOT NULL,
+  model_number TEXT NOT NULL,
+  PRIMARY KEY (id),
+  UNIQUE (model_number),
+  FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE CASCADE
+);
```

Comment:
We should define the scope of `frobulators` up front here. All resources in Coder need to live in at least 1 bucket. Site, Organization, and/or User.

In `TestRolePermissions` a `frobulator` is defined as: 
```golang
rbac.ResourceFrobulator.WithOwner(currentUser.String()).InOrg(orgID)
```

In `dbauthz`, `GetUserFrobulators` defines one as
```golang
rbac.ResourceFrobulator.WithOwner(userID.String())
```

These are not the same. The first belongs to an organization **and** a user. The second belongs to just the user. 

Meaning without `InOrg(orgID)`, organization roles have no authority over said resource.

So if we intend to org scope this resource (which is usually the case), we should also add an `organization_id` column.

----

Almost all our database objects have a `RBACObject()` method that handles this ownership scoping. [See workspace](https://github.com/coder/coder/blob/ceffff9b0676ef708e6ccf40d24279140824e670/coderd/database/modelmethods.go#L169-L178). 

I agree though that this is purely a enforcement by PR reviews, there is no code mechanism to make sure you have inputted the correct fields except the unit test in `dbauthz`. It is not defined in the policy :cry: